vowpalwabbit/csv_parser/src/parse_example_csv.cc

```diff
@@ -0,0 +1,517 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#include "vw/csv_parser/parse_example_csv.h"
+
+#include "vw/core/best_constant.h"
+#include "vw/core/parse_args.h"
+#include "vw/core/parse_primitives.h"
+#include "vw/core/parser.h"
+
+#include <string>
+
+namespace VW
+{
+namespace parsers
+{
+int parse_csv_examples(VW::workspace* all, io_buf& buf, VW::multi_ex& examples)
+{
+  bool keep_reading = all->custom_parser->next(*all, buf, examples);
+  return keep_reading ? 1 : 0;
+}
+
+void csv_parser::handling_csv_separator(std::string& str, const std::string& name)
```

Comment:
since this method has side effects (i.e. might change the separator passed in as an argument) it would be nice to have the methods name declare its intentions e.g.
```suggestion
void csv_parser::set_csv_separator(std::string& str, const std::string& name)
```

Reply:
Fixed!